File: scripts/generate_rsa_keys.py
#!/usr/bin/env python3

import subprocess


def run():
    try:
        subprocess.run(
            ["openssl", "genrsa", "-out", "og_private.pem", "2048"],
            capture_output=True,
            text=True,
            check=True,
        )

        subprocess.run(
            [
                "openssl",
                "rsa",
                "-in",
                "og_private.pem",
                "-outform",
                "PEM",
                "-pubout",
                "-out",
                "og_public.pem",
            ],
            capture_output=True,
            text=True,
            check=True,
        )

    except Exception as e:
        print(e)


if __name__ == "__main__":
    run()

File: scripts/generate_rsa_keys.js
#!/usr/bin/env nodejs

const crypto = require('crypto');
const fsPromises = require('fs').promises;

function genKeys() {
  return new Promise((resolve, reject) => {
    crypto.generateKeyPair('rsa', {
      modulusLength: 2048,
      publicKeyEncoding: {
        type: 'spki',
        format: 'pem',
      },
      privateKeyEncoding: {
        type: 'pkcs1',
        format: 'pem'
      },
    }, (error, publicKey, privateKey) => {
      if (error) {
        return reject(error);
      }

      return resolve([publicKey, privateKey]);
    });
  });
}

async function run() {
  const [publicKey, privateKey] = await genKeys();

  await fsPromises.writeFile('og_public.pem', publicKey);
  await fsPromises.writeFile('og_private.pem', privateKey);
}

run().then(() => console.log('Keys generated'));
